presto-hive/src/main/java/com/facebook/presto/hive/EmptyFileCreator.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive;
+
+import com.facebook.presto.hive.metastore.Partition;
+import com.facebook.presto.hive.metastore.Table;
+import com.facebook.presto.spi.ConnectorSession;
+import org.apache.hadoop.fs.Path;
+
+import java.util.List;
+import java.util.Optional;
+
+public interface EmptyFileCreator
+{
+    void createFiles(ConnectorSession session, Path path, Table table, Optional<Partition> partition,
```

Comment:
This method signature can be quite confusing. Mostly because the infuses two different concepts. It is supposed to create zero row files at the given location, but it takes high level logical objects, such as `Table` and `Partition`.

I would recommend changing the signature to something like

```
void create(
  ConnectorSession session,
  HdfsContext hdfsContext,
  Path destinationDirectory,
  List<String> fileNames,
  StorageFormat format,
  HiveCompressionCodec compressionCodec, 
  Properties schema
)
```